sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -243,13 +243,61 @@ namespace Azure { namespace Core { namespace Test {
           "Test Log from: [ " + m_testContext.GetTestPlaybackRecordingName() + " ] - " + message);
     }
 
-    // Util for tests getting env vars
-    std::string GetEnv(const std::string& name)
+    /**
+     * @brief Utility function used by tests to retrieve env vars
+     *
+     * @param name Environment variable name to retrieve.
+     *
+     * @return The value of the environment variable retrieved.
+     *
+     * @note If AZURE_TENANT_ID, AZURE_CLIENT_ID, or AZURE_CLIENT_SECRET are not available in the
+     * environment, the AZURE_SERVICE_DIRECTORY environment variable is used to set those values
+     * with the values emitted by the New-TestResources.ps1 script.
+     *
+     * @note The Azure CI pipeline upper cases all environment variables defined in the pipeline.
+     * Since some operating systems have case sensitive environment variables, on debug builds, this
+     * function ensures that the environment variable being retrieved is all upper case.
+     *
+     */
+    std::string GetEnv(std::string const& name)
     {
-      const auto ret = Azure::Core::_internal::Environment::GetVariable(name.c_str());
-
+#if !defined(NDEBUG)
+      // The azure CI pipeline uppercases all EnvVar values from ci.yml files.
+      // That means that any mixed case strings will not be found when run from the CI
+      // pipeline. Check to make sure that the developer only passed in an upper case environment
+      // variable.
+      {
+        if (name != Azure::Core::_internal::StringExtensions::ToUpper(name))
+        {
+          throw std::runtime_error("All Azure SDK environment variables must be all upper case.");
```

Comment:
Here's an idea. It can be done after this PR anyway.

Instead of throw if `name` is not all UPPERCASE, let's always call `ToUpper(name)` and document that the c++ test-fw do the same as the ADO and works with upper case env vars only.

We are "kind" of doing something like this already with `AZURE_TEST_MODE`, where the value is checked regardless of the case: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core-test/src/interceptor_manager.cpp#L29

Reply:
That's an interesting suggestion - my biggest concern with it is that penalizes developers who provide their names in uppercase always. My change was intentionally only on debug builds to reduce the performance consequences (otoh, we're about to spend many milliseconds talking to a remote service, so....)

Reply:
Created Issue #3375 to track this suggestion.

Reply:
It would be only for test-fw.